src/components/tl/ucp/tl_ucp.c

```diff
@@ -110,11 +116,17 @@ static ucc_config_field_t ucc_tl_ucp_lib_config_table[] = {
      UCC_CONFIG_TYPE_UINT},
 
     {"REDUCE_AVG_PRE_OP", "1",
-     "Reduce will perform division by team_size in early stages of the algorithm,\n"
+     "Reduce will perform division by team_size in early stages of the "
+     "algorithm,\n"
      "else - in result",
      ucc_offsetof(ucc_tl_ucp_lib_config_t, reduce_avg_pre_op),
      UCC_CONFIG_TYPE_BOOL},
 
+    {"REDUCE_SCATTER_RING_BIDIRECTIONAL", "y",
```

Comment:
do you have some perf data of unidirectional vs bidirectional? should we add bidirectional implementation for other ring algorithms e.g. allgather?

Reply:
added to PR desc. Maybe it worth extending ag as well at some point. It is not always beneficial (e.g., if ring covers CPU SHM then there is no gain, since shm is not bibw).